geoportailv3/static/js/locationinfo/locationinfodirective.js

```diff
@@ -0,0 +1,278 @@
+   * @private
+   */
+  this.coordinateString_ = appCoordinateString;
+
+  /**
+   * @type {ol.FeatureOverlay}
+   * @private
+   */
+  this.featureOverlay_ = this.createFeatureOverlay_();
+
+  /**
+   * @type {boolean}
+   */
+  this['featureOverlayItemExists'] = false;
+
+  $scope.$watch(goog.bind(function() {
+    return this['appSelector'];
+  }, this), goog.bind(function(newVal) {
+    if (newVal != 'locationinfo') {
+      var features = this.featureOverlay_.getFeatures();
+      features.clear();
+    }
+  }, this));
+
+  $scope.$watch(goog.bind(function() {
+    return this['open'];
+  }, this), goog.bind(function(newVal) {
+    var features = this.featureOverlay_.getFeatures();
+    if (newVal === false) {
+      this['appSelector'] = undefined;
+      features.clear();
+      this['featureOverlayItemExists'] = false;
+    } else if (newVal === true && features.getLength() >= 1) {
+      this['featureOverlayItemExists'] = true;
+    }
+  }, this));
+
+  /**
+   * @type {string}
+   */
+  this['url'] = '';
+
+  /**
+   * @type {string}
+   */
+  this['qrUrl'] = '';
+
+  /**
+   * @type {string}
+   */
+  this['elevation'] = '';
+
+  /**
+   * @type {app.GetElevation}
+   * @private
+   */
+  this.getElevation_ = appGetElevation;
+
+  /**
+   * @type {app.GetShorturl}
+   * @private
+   */
+  this.getShorturl_ = appGetShorturl;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.qrServiceUrl_ = qrServiceUrl;
+
+  /**
+   * @type {Object}
+   * @private
+   */
+  this.projections_ = {
+    'EPSG:2169': 'Luref',
+    'EPSG:4326': 'Lon/Lat WGS84',
+    'EPSG:4326:DMS': 'Lon/Lat WGS84 DMS',
+    'EPSG:3263*': 'WGS84 UTM'
+  };
+
+  /**
+   * @type {ol.Coordinate|undefined}
+   */
+  this['coordinate'] = undefined;
+
+  /**
+   * @type {Object}
+   */
+  this['location'] = {};
+
+  $scope.$watch(goog.bind(function() {
+    return this['coordinate'];
+  }, this), goog.bind(function(newVal) {
+    if (goog.isDef(newVal)) {
+      this['location'] = {};
+      goog.object.forEach(this.projections_, function(value, key) {
+        var sourceEpsgCode = this['map'].getView().getProjection().getCode();
+        if (key === 'EPSG:4326:DMS') {
+          var epsgCode = goog.string.remove(key, ':DMS');
+          this['location'][value] = this.coordinateString_(
+              this['coordinate'], sourceEpsgCode, epsgCode, true);
+        } else {
+          this['location'][value] = this.coordinateString_(
+              this['coordinate'], sourceEpsgCode, key);
+        }
+      }, this);
+    }
+  }, this));
+
+  this['map'].getViewport()
+    .addEventListener('contextmenu', goog.bind(function(event) {
+        event.preventDefault(); // disable right-click menu on browsers
+      }, this));
+
+  /**
+   * @type {angular.$timeout}
+   * @private
+   */
+  this.$timeout_ = $timeout;
+
+  /**
+   * @type {?Object<number, number>}
+   * @private
+   */
+  this.startPixel_ = null;
+
+  /**
+   * @type {angular.$q.Promise|undefined}
```

Comment:
I tried that last week, but wouldnt pass compiler, as `this.$timeout_.cancel(this.holdPromise_)` expects `undefined`, not `null`:

````
ERR! compile /home/wepa/geoportailv3/geoportailv3/static/js/locationinfo/locationinfodirective.js:194: ERROR - actual parameter 1 of this.$timeout_.cancel does not match formal parameter
ERR! compile found   : (null|{catch: function ((function (?): ?|null)): ?, finally: function ((function (?): ?|null)): ?, then: function ((function (?): ?|null), (function (?): ?|null)=, (function (?): ?|null)=): ?})
ERR! compile required: (angular.$q.Promise|undefined)
ERR! compile           this.$timeout_.cancel(this.holdPromise_);
ERR! compile                                 ^
ERR! compile
ERR! compile /home/wepa/geoportailv3/geoportailv3/static/js/locationinfo/locationinfodirective.js:205: ERROR - actual parameter 1 of this.$timeout_.cancel does not match formal parameter
````